src/workerinfo.js

```diff
@@ -34,57 +35,65 @@ class WorkerInfo {
     }
   }
 
-  async seen(provisionerId, workerType) {
+  async seen(provisionerId, workerType, workerGroup, workerId) {
     const expired = entity => Date.now() - new Date(entity.expires) > 24 * 60 * 60 * 1000;
     const expires = taskcluster.fromNow('5 days');  // temporary hard coded expiration
 
+    const updateExpiration = (entry) => {
+      entry.modify(entity => {
+        if (expired(entity)) {
+          entity.expires = expires;
+        }
+      });
+    };
+
+    const createEntry = async (entity, entry) => {
```

Comment:
this is nit... but I would suggest a global utility like:
```js
let ignoreEntityAlreadyExists = (err) => {
  if (!err || err.code !== 'EntityAlreadyExists') {
    throw err;
  }
}
```

Then use it like:
```js
await this.workerType.create({provisionerId, workerType, expires}).catch(ignoreEntityAlreadyExists);
```